### PR TITLE
memory game fact list refactored

### PR DESCRIPTION
**Pull request by tmansuroglu**
#22 Part of refactored final code.

### CODE REVIEW

File: src/components/memorygame/MemoryGameFactList/index.jsx [hunk 1]
```diff
@@ -1,31 +1,65 @@
...
+ * If the language is English "emptyFactListMessage" will be ""Match cards to reveal facts!"
  */
-const MemoryGameFactList = ({ facts, title }) => {
-  return (
-    <>
-      <h1 className="memoryGame__title">{title}</h1>
+const MemoryGameFactList = ({ facts }) => {
+  // facts example:
+  // {
+  //   cardKey: "alternateEnergy0",
```
Comment (louisrli) [inline src/components/memorygame/MemoryGameFactList/index.jsx]: again i dont recommend doing soemthing like this in futrue projects but its fine for now. generally you'd want to use something like typescript to define an interface. the problem with this approach is that its very brittle to code changes, there is nothing stopping you from adding/removnig a field from facts and forgetting that this comment exists.

File: src/components/memorygame/MemoryGameFactList/index.jsx [hunk 2]
```diff
@@ -1,31 +1,65 @@
...
+  //   description: "Alternatif Enerji",
+  //   img: {src: "/static/media/alternateEnergy.88cef00e.svg", imgKey: "alternateEnergy"},
+  //   isFlipped: false,
+  //   isMatched: false,
+  //   link: "https://loremIpsum.com",
+  //   phrase: "Lorem ipsum",
+  // }
+  const [t] = useTranslation();
+  const factsTitle = t("memoryGame.facts");
+  const emptyFactListMessage = t("memoryGame.emptyFactListMessage");
```
Comment (louisrli) [inline src/components/memorygame/MemoryGameFactList/index.jsx]: i would just inline lines 27 to 31. a bit too overboard with the variables, they are perfectly readable if they are inline / they are not being used more than once. i'd say the code is catually harder to read when it's here

File: src/components/memorygame/MemoryGameFactList/index.jsx [hunk 3]
```diff
@@ -1,31 +1,65 @@
...
+  const factListTitle = (
+    <h1 className="memoryGameFactListContainer__title">{factsTitle}</h1>
+  );
+  let factListContent;
+  // facts is an empty array by default, it gets filled when user makes successful match
+  if (facts.length) {
```
Comment (louisrli) [inline src/components/memorygame/MemoryGameFactList/index.jsx]: facts.length !== 0, be explicit, you may think it's shorter and fancy but it takes the reader a second to think about this

**Review by louisrli — CHANGES_REQUESTED** [review]

File: src/components/memorygame/MemoryGameFactList/index.jsx (hunk 1 above)
Comment (tmansuroglu) [reply src/components/memorygame/MemoryGameFactList/index.jsx]: PTAL

File: src/components/memorygame/MemoryGameFactList/index.jsx (hunk 2 above)
Comment (tmansuroglu) [reply src/components/memorygame/MemoryGameFactList/index.jsx]: PTAL

File: src/components/memorygame/MemoryGameFactList/index.jsx (hunk 3 above)
Comment (tmansuroglu) [reply src/components/memorygame/MemoryGameFactList/index.jsx]: PTAL

**Review by louisrli — APPROVED** [review]

**Review by ssinikoglu — APPROVED** [review]
Nice description and code, I don't have any comments to add.

**Review by ibrahimaljarad — APPROVED** [review]

**Comment by tmansuroglu** [comment]
@Ammar-64  can you merge?

**Pull request merged**